elixir/apps/domain/test/domain/accounts_test.exs

```diff
@@ -583,6 +583,118 @@ defmodule Domain.AccountsTest do
              }
     end
 
+    test "returns error when search_domain is invalid", %{account: account} do
+      attrs = %{
+        config: %{
+          search_domain: "invalid"
+        }
+      }
+
+      assert {:error, changeset} = update_account_by_id(account.id, attrs)
+
+      assert errors_on(changeset) == %{
+               config: %{
+                 search_domain: ["must be a valid fully-qualified domain name"]
+               }
+             }
+    end
+
+    test "returns error when search_domain is too long", %{account: account} do
+      attrs = %{
+        config: %{
+          search_domain: String.duplicate("a", 256)
+        }
+      }
+
+      assert {:error, changeset} = update_account_by_id(account.id, attrs)
+
+      assert errors_on(changeset) == %{
+               config: %{
+                 search_domain: ["must not exceed 255 characters"]
+               }
+             }
+    end
+
+    test "returns error when search_domain starts with a dot", %{account: account} do
+      attrs = %{
+        config: %{
+          search_domain: ".example.com"
+        }
+      }
+
+      assert {:error, changeset} = update_account_by_id(account.id, attrs)
+
+      assert errors_on(changeset) == %{
+               config: %{
+                 search_domain: ["must not start or end with a dot"]
+               }
+             }
+    end
+
+    test "returns error when search_domain ends with a dot", %{account: account} do
+      attrs = %{
+        config: %{
+          search_domain: "example.com."
```

Comment:
That is technically a valid domain.

Reply:
Hmm, yeah I suppose we should them to end with a dot.

Reply:
#8392